BFD-3471: Update PR template (#2373)
CMSgov/beneficiary-fhir-data: .github/pull_request_template.md

**JIRA Ticket:**

> [BFD-123456](https://jira.cms.gov/browse/BFD-123456)

Reviewer: Consider simplifying this. This repository is configured to link DASG projects, e.g. `BFD-123` -> becomes BFD-123 without the markdown. I'll leave that up to as to whether we remove the markdown formatting for the jira issue link 😉 
```suggestion
BFD-XXXX
```

-->

If you're reviewing this PR, please check for these things in particular:
<!-- Add some items here -->

### What Security Implications Does This PR Have?

Please indicate if this PR does any of the following:  

* Adds any new software dependencies
* Modifies any security controls
* Adds new transmission or storage of data

> * Any other changes that could possibly affect secuirity? 

Reviewer: ```suggestion
* Any other changes that could possibly affect security? 
```

> * [ ] Yes, one or more of the above security implications apply. This PR must not be merged without the ISSO or team security engineer's (@sb-benohe) approval. 

Reviewer: I'd like to avoid check boxes that we don't expect to regularly be checked–github treats unchecked boxes as incomplete tasks in the PR interface. Could we do something like the following instead?
```suggestion
* [ ] I have considered the above security implications as it relates to this PR. (If one or more of the above apply, it cannot be merged without the ISSO or team security engineer's (@sb-benohe) approval.) 
```
